### DISCUSSION TITLE
some question about “TransportOrder”

### DISCUSSION THREAD

**Discussion by yaowuccheng** [openTCS/opentcs#7] [Q&A] [answered]
how to pause the transportorder when the vehicle arrive on  workstation？i use ‘ kernelExecutor.wait()’  for pause the transportorder and it paused， i use ‘kernelExecutor.notify()’  and it start .But it works for all vehicles，how can let it works for only a vehicle？ Thanks！！

**Comment by swltr** [comment, marked as answer]
`wait()` on an `ExecutorService` is inherited from `Object`. Calling it would pause the thread in which you call it, so the resulting behaviour depends on what that thread does.

Strictly speaking, you cannot pause a transport order. But you could pause a vehicle before/after/as part of executing the destination operation. Either the vehicle itself would have to pause, e.g. by waiting for someone to press a button to confirm some manual interaction, or the vehicle driver would have to delay calling `commandExecuted()` on its model object.
